django/api/tests.py

```diff
@@ -77,3 +79,114 @@ def test_restaurants_default_order_by_name(self):
+        self.assertEqual(2, len(restaurants))
+
+    def test_user_is_owner(self):
+        user = User.objects.get(username=self.test_user.username)
+        user_data = UserSerializer(user).data
+        restaurants = Restaurant.objects.filter(owner=user)
+        is_owner = True
+        for restaurant in restaurants:
+            owner = dict(
+                RestaurantSerializer(restaurant).data["properties"].pop("owner")[0]
+            )
+            if owner != user_data:
+                is_owner = False
+        self.assertEqual(True, is_owner)
+
+
+class UserViewsetTestCase(TestCase):
+    def setUp(self):
+        self.c = Client()
+        self.userdata_owner_1 = {
+            "user": {"username": "testOwner1", "password": "testOwner1234"},
+            "restaurant_name": "testOwner1s Pizza Joint",
+            "restaurant_address": "1 Pizza Plaza Philadelphia, PA 12345",
+            "restaurant_loc": {"type": "Point", "coordinates": [125.6, 10.1]},
+        }
+        self.userdata_owner_2 = {
+            "user": {"username": "testOwner2", "password": "testOwner1234"},
+            "restaurant_name": "testOwner2s Pizza Joint",
+            "restaurant_address": "2 Pizza Plaza Philadelphia, PA 12345",
+            "restaurant_loc": {"type": "Point", "coordinates": [125.6, 10.1]},
+        }
+        self.userdata_not_owner = {
+            "user": {"username": "testNotAnOwner1", "password": "testOwner1234"}
+        }
+
+    def test_user_creation_not_owner(self):
+        response = self.c.post(
+            "/api/user/", self.userdata_not_owner, content_type="application/json"
+        )
+        # check created status
+        self.assertEqual(response.status_code, 201)
+        # check user exists
+        does_user_exist = User.objects.filter(
+            username=self.userdata_not_owner["user"]["username"]
+        ).exists()
+        self.assertEqual(True, does_user_exist)
+
+    def test_user_creation_is_owner(self):
+        response = self.c.post(
+            "/api/user/", self.userdata_owner_1, content_type="application/json"
+        )
+        # check created status
+        self.assertEqual(response.status_code, 201)
+        # check user exists
+        does_user_exist = User.objects.filter(
+            username=self.userdata_owner_1["user"]["username"]
+        ).exists()
+        self.assertEqual(True, does_user_exist)
+        # check restaurant exists
+        does_restaurant_exist = Restaurant.objects.filter(
+            name=self.userdata_owner_1["restaurant_name"]
+        ).exists()
+        self.assertEqual(True, does_restaurant_exist)
+
+    def test_restaurant_owner_field_set(self):
+        self.c.post(
+            "/api/user/", self.userdata_owner_2, content_type="application/json"
+        )
+        restaurant = Restaurant.objects.get(
+            name=self.userdata_owner_2["restaurant_name"]
+        )
+        restaurant_owner = dict(
```

Comment:
I'm not sure that it's necessary to re-serialize the objects you get from the database in this way before testing. It makes sense that you are posting to the API, and then querying the database via a different route to verify that the created objects match what you expect, but I think it should be sufficient to simply compare the `.pk` property of each candidate `User` object; if the primary keys match then you know you've got the same object. But I might not be understanding what's happening here correctly--what do you think?